smartlearner/interfaces/model.py

```diff
@@ -30,8 +22,15 @@ def parameters(self):
 
     @abstractmethod
     def save(self, path):
+        """ Saves model information to disk. """
         raise NotImplementedError("Subclass of 'Model' must implement 'save(path)'.")
 
-    @abstractclassmethod
+    @abstractmethod
     def load(self, path):
+        """ Loads model information from disk. """
         raise NotImplementedError("Subclass of 'Model' must implement 'load(path)'.")
+
+    @classmethod
+    def create(cls, path):
```

Comment:
The class method `create` first creates a new instance of the model using the hyperparameters found in `path` (probably a `.json` file), then it sets the parameters of this new instance (via the `load` function) according to the parameters saved in `path`(probably a `.npy` file). Finally, it returns the new instance.

Versus the `load` method that simply sets the parameters that have been saved in `path`.

When resuming a training, we first construct the 'trainer's graph' (i.e. we instantiate the model, the batch_scheduler, the optimizer, the trainer and all the tasks.), then we called `trainer.load` to bring back the training state (i.e. replaces parameters, if needed, of the model, the optimizer, the batch_scheduler, the trainer and the task).